labapi/src/main/java/com/microsoft/identity/internal/test/labapi/ApiClient.java

```diff
@@ -854,8 +855,25 @@ public <T> ApiResponse<T> execute(Call call, Type returnType) throws ApiExceptio
             Response response = call.execute();
             T data = handleResponse(response, returnType);
             return new ApiResponse<T>(response.code(), response.headers().toMultimap(), data);
-        } catch (IOException e) {
-            throw new ApiException(e);
+        } catch (IOException exception) {
+            // If we receive an IO Exception, we should retry once, to avoid inconsistent
+            // network causing test failures. The timeout has been reduced to allow this.
+
+            try {
+                // Wait for a bit
+                try {
+                    Thread.sleep(LAB_API_RETRY_WAIT);
+                } catch (final InterruptedException e2) {
+                    e2.printStackTrace();
+                }
+                Response response = call.execute();
+                T data = handleResponse(response, returnType);
+                return new ApiResponse<T>(response.code(), response.headers().toMultimap(), data);
+            } catch (IOException secondException) {
+                // If we get another IO Exception, we should fail. Most likely another retry
+                // will not resolve the issue.
+                throw new ApiException(secondException);
+            }
```

Comment:
It seems we are just repeating the code in 855-857.

Instead could consider using recursion and with an n value that terminates the recursion once n reaches 2 in this case i.e. 2 io exceptions